crates/autopilot/src/limit_orders/metrics.rs

```diff
@@ -14,43 +15,47 @@ pub struct LimitOrderMetrics {
-                let awaiting_quote = self
-                    .database
-                    .count_limit_orders_with_outdated_fees(self.quoting_age)
-                    .await
-                    .unwrap();
-                let unusable = self
-                    .database
-                    .count_limit_orders_with_outdated_fees(self.validity_age)
-                    .await
-                    .unwrap();
-
-                let quoted_limit_orders = limit_orders - awaiting_quote;
-                let usable_limit_orders = limit_orders - unusable;
-
-                limit_orders_gauge.set(limit_orders);
-                unquoted_limit_orders_gauge.set(awaiting_quote);
-                quoted_limit_orders_gauge.set(quoted_limit_orders);
-                usable_limit_orders_gauge.set(usable_limit_orders);
-                unusable_limit_orders_gauge.set(unusable);
-
-                tokio::time::sleep(Duration::from_secs(10)).await;
+        tokio::spawn(
+            async move {
+                let limit_orders_gauge = gauge("limit_orders", "Open limit orders.");
+                let quoted_limit_orders_gauge =
+                    gauge("quoted_limit_orders", "Quoted limit orders.");
+                let unquoted_limit_orders_gauge =
+                    gauge("unquoted_limit_orders", "Limit orders awaiting a quote.");
+                let usable_limit_orders_gauge =
+                    gauge("usable_limit_orders", "Limit orders usable in the auction.");
+                let unusable_limit_orders_gauge = gauge(
+                    "unusable_limit_orders",
+                    "Limit orders with surplus_fee too old for the auction.",
+                );
+
+                loop {
+                    let limit_orders = self.database.count_limit_orders().await.unwrap();
+                    let awaiting_quote = self
+                        .database
+                        .count_limit_orders_with_outdated_fees(self.quoting_age)
+                        .await
+                        .unwrap();
+                    let unusable = self
+                        .database
+                        .count_limit_orders_with_outdated_fees(self.validity_age)
+                        .await
+                        .unwrap();
+
+                    let quoted_limit_orders = limit_orders - awaiting_quote;
+                    let usable_limit_orders = limit_orders - unusable;
+
+                    limit_orders_gauge.set(limit_orders);
+                    unquoted_limit_orders_gauge.set(awaiting_quote);
+                    quoted_limit_orders_gauge.set(quoted_limit_orders);
+                    usable_limit_orders_gauge.set(usable_limit_orders);
+                    unusable_limit_orders_gauge.set(unusable);
+
+                    tokio::time::sleep(Duration::from_secs(10)).await;
+                }
             }
-        });
+            .instrument(tracing::info_span!("LimitOrderMetrics")),
```

Comment:
naming-nit. We've been using `snake_case` for tracing spans. Might be worth coming up with a standard for the span names so they are consistent everywhere (doesn't have to be `snake_case`)

Reply:
I started out using `snake_case` and stumbled over [this](https://github.com/cowprotocol/services/blob/main/crates/autopilot/src/lib.rs#L561-L565) which used `CamelCase`. I guess I can just turn everything to snake case including the existing info span.

Reply:
Not necessarily set on `snake_case` (although, this would likely be my subjective preference). However, there might even be an argument for `PascalCase` or `SCREAMING_SNAKE_CASE` so that it stands out from the log module name. Up to you on what you think would be best in this case.

> I guess I can just turn everything to snake case including the existing info span.

Personally, I like this idea.